### ISSUE TITLE
Can shallow fusion support arbitrary language models other than RNNLM?

### ISSUE THREAD

**Issue by getao** [OpenNMT/OpenNMT#507]
Hello. I wonder if I can use arbitrary language models (e.g., an n-gram LM) to help decoding using the shallow fusion strategy. It seems that currently OpenNMT only supports shallow fusion with RNNLM trained by itself.

Thank you in advance.

**Comment by jsenellart** [comment]
yes - as of today, we only support onmt rnnlm since code is integrated. through hooks mechanism it would not be hard to extend to other LM though. Do you have one in particular in mind?

**Comment by getao** [comment]
Thanks @jsenellart . I'll have a look at the hook mechanism.
